sensors/simsensor_common/simsensor_reader.c

```diff
@@ -0,0 +1,280 @@
+/*
+ * Phoenix-RTOS
+ *
+ * Simsensor reader reads and parses data form special scenario file
+ * than it pushes it to event_queue
+ *
+ * Copyright 2023 Phoenix Systems
+ * Author: Piotr Nieciecki
+ *
+ * This file is part of Phoenix-RTOS.
+ *
+ * %LICENSE%
+ */
+
+#include <stdio.h>
+#include <stdlib.h>
+#include <ctype.h>
+
+#include "simsensor_reader.h"
+
+#define READER_END_SCENARIO_INDICATOR NO_SENSOR
+#define READER_DATA_SEPARATOR         ','
+
+#define READER_TIMESTAMP_NOT_SET -1
+
+#define READER_ACCEL_DEV_ID 1
+#define READER_BARO_DEV_ID  2
+
+
+typedef struct {
+	time_t first_scenario_time;
+	time_t last_scenario_timestamp;
+	int loops_cnt;
+} simsens_time_t;
+
+
+struct simsens_reader_t {
+	FILE *scenario_file;
+	int sensor_types;
+
+	time_t time_horizon;
+
+	simsens_time_t time_struct;
+};
+
+
+static void reader_timeInit(simsens_time_t *timeStruct)
+{
+	timeStruct->first_scenario_time = READER_TIMESTAMP_NOT_SET;
+	timeStruct->last_scenario_timestamp = READER_TIMESTAMP_NOT_SET;
+	timeStruct->loops_cnt = 0;
+}
```

Comment:
Implementing this function directly in the `init` does not impact clarity of code much. Let`s use it there